lib/translator/command_translator.rb

```diff
@@ -117,8 +128,21 @@ def visit_array_new_command(command)
                 kernel_builder.add_methods(block_translation_result.aux_methods)
                 kernel_builder.add_block(block_translation_result.block_source)
 
+                if command.keep
```

Comment:
This `if` check and the code that is within keep appearing multiple times. I think we can create an instance method in here named `build_command_translation_result(execution:, result:, return_type:, keep: false)' that we call instead and creates the `CommandTranslationResult`. If `keep` is `true`, then it also adds the cached result etc.